tests/integration/components/polaris-choice-list-test.js

```diff
@@ -1,586 +1,599 @@
+
+  hooks.beforeEach(function(/* assert */) {
+    this.owner.register('component:svg-jar', MockSvgJarComponent);
+  });
+
+  test('it renders the correct HTML when allowMultiple is false', async function(assert) {
+    await render(hbs`
+      {{polaris-choice-list
+        name="test-single-choice-list"
+        allowMultiple=false
+        choices=(array
+          (hash
+            label="First option"
+            value="one"
+          )
+          (hash
+            label="Second option"
+            value="two"
+          )
+          (hash
+            label="Third option"
+            value="three"
+          )
         )
-        (hash
-          label="Third option"
-          value="three"
+        selected=(array
+          "two"
         )
-      )
-      selected=(array
-        "two"
-      )
-    }}
-  `);
-
-  const choices = findAll(choiceSelector);
-  assert.equal(choices.length, 3, 'renders three choices');
-
-  // Check the choices.
-  let choice = choices[0];
-  assert.equal(
-    choice.textContent.trim(),
-    'First option',
-    'first choice - has correct label text'
-  );
-  let radioInput = find(radioInputSelector, choice);
-  assert.ok(radioInput, 'first choice - renders radio input');
-  assert.equal(
-    radioInput.value,
-    'one',
-    'first choice - radio input has the correct value'
-  );
-  assert.equal(
-    radioInput.name,
-    'test-single-choice-list',
-    'first choice - radio input has the correct name'
-  );
-  assert.notOk(radioInput.checked, 'first choice - radio input is not checked');
-
-  choice = choices[1];
-  assert.equal(
-    choice.textContent.trim(),
-    'Second option',
-    'second choice - has correct label text'
-  );
-  radioInput = find(radioInputSelector, choice);
-  assert.ok(radioInput, 'second choice - renders radio input');
-  assert.equal(
-    radioInput.value,
-    'two',
-    'second choice - radio input has the correct value'
-  );
-  assert.equal(
-    radioInput.name,
-    'test-single-choice-list',
-    'second choice - radio input has the correct name'
-  );
-  assert.ok(radioInput.checked, 'second choice - radio input is checked');
-
-  choice = choices[2];
-  assert.equal(
-    choice.textContent.trim(),
-    'Third option',
-    'third choice - has correct label text'
-  );
-  radioInput = find(radioInputSelector, choice);
-  assert.ok(radioInput, 'third choice - renders radio input');
-  assert.equal(
-    radioInput.value,
-    'three',
-    'third choice - radio input has the correct value'
-  );
-  assert.equal(
-    radioInput.name,
-    'test-single-choice-list',
-    'third choice - radio input has the correct name'
-  );
-  assert.notOk(radioInput.checked, 'third choice - radio input is not checked');
-});
+      }}
+    `);
+
+    const choices = findAll(choiceSelector);
+    const radioInputs = findAll(radioInputSelector);
+    assert.equal(choices.length, 3, 'renders three choices');
+
+    // Check the choices.
+    let choice = choices[0];
```

Comment:
no need to use these anymore, you can just use `assert.dom(`${choiceSelector}:nth-child(1))` to target first item and so on

Reply:
The `nth` selectors don't work in these cases because the choice selectors aren't all children of the same parent, so the test will only ever find `nth-child(1)` or `nth-of-type(1)` and then bork on trying to find the 2nd or 3rd, etc.